.github/workflows/flow-micro-benchmarks.yml

```diff
@@ -0,0 +1,122 @@
+name: Run a Micro Benchmark Flow
+
+on:
+    # Manual trigger
+    workflow_dispatch:
+      inputs:
+        profile_env:
+          type: number # for default of 0
+        rejson_branch:
+          type: string
+          default: master
+          description: 'branch to use for rejson'
+        benchmark_folder: 
+          type: string
+          default: src/redisearch_rs/trie_bencher
+          description: folder containing the benchmark to be run with `cargo bench`
+        output_folder:
+          type: string
+          default: bin/criterion/trie
+          description: folder used to store the output of criterion
+        memory_limit:
+          type: number
+          default: 12582912 # that's 12GB
+          description: Used by the memory monitoring to understand if a runner terminates because of lack of memory (leaking)
+    
+    # For calling from other workflows
+    workflow_call:
+      inputs:
+        profile_env:
+          type: number # for default of 0
+        rejson_branch:
+          type: string
+          default: master
+          description: 'branch to use for rejson'
+        benchmark_folder: 
+          type: string
+          default: src/redisearch_rs/trie_bencher
+          description: folder containing the benchmark to be run with `cargo bench`
+        output_folder:
+          type: string
+          default: bin/criterion/trie
+          description: folder used to store the output of criterion
+        memory_limit:
+          type: number
+          default: 12582912 # that's 12GB
+          description: Used by the memory monitoring to understand if a runner terminates because of lack of memory (leaking)
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-24.04
+    env:
+      RUST_BACKTRACE: full
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Setup specific  # assuming node20 supported
+        working-directory: .install
+        run: |
+          ./install_script.sh ${{ steps.mode.outputs.mode }} sudo
+      - name: Setup tests dependencies
+        run: |
+          .install/test_deps/common_installations.sh ${{ steps.mode.outputs.mode }} sudo
```

Comment:
Do you want to hard-code the mode to `sudo` or do you want to compute it in a step upstream? We need to pick one strategy here.